src/sage/combinat/words/word.py

```diff
@@ -659,10 +661,11 @@ class Word_iter_with_caching(WordDatatype_iter_with_caching, Word_class):
 
     Pickle is not supported for word of unknown length defined by an iterator::
 
-        sage: dumps(w)
-        Traceback (most recent call last):
-        ...
-        TypeError: can...t...pickle...generator...object...
+        sage: try:
+        ....:     dumps(w)
+        ....: except TypeError as e:
+        ....:     "pickle" in str(e) and "generator" in str(e)
```

Comment:
can this be made more clear - instead of merely testing a condion, make it print a meaningful
message if it's True:
```
....:     if "pickle" in str(e) and "generator" in str(e):
....:         print("TypeError rised in dumps()")
TypeError rised in dumps()
```

Reply:
typos - `s/rised/raised` (2 times)